api-server/README.md

```diff
@@ -0,0 +1,112 @@
+
+For example to understand what problem the API server solves, the node software only stores blocks of the blockchain, but it does not index the transactions by their id. Meaning: Attempting to find a transaction by its id is virtually impossible without going through all blocks. This same applies to more information that's not directly, minimally, required to operate the blockchain. In that case, using the API server solves all these problems, since the API server is made to index the information and put it in the database.
+
+## Architecture
+
+#### The API web server
+
+The API web server contains the restful endpoints that can be reached by the client application, such as the block explorer or the web wallet. The API web server communicates with the database in the backend to respond to queries, in addition to the optional possibility to communicate with the node for specialized requests, such as submitting transactions or finding out the current transaction fee in the mempool.
+
+#### The blockchain scanner daemon
+
+The blockchain scanner daemon is a tool that runs in the backend, scans the blockchain, and updates the database information.
+
+#### How the API web server and the blockchain scanner daemon work together
+
+The architecture of the API server is made to be distributed as much as desired. You can run the database on as many servers as you wish in master-slave mode. This is achieved by separating the "API web server" from the "blockchain scanner daemon". You can have a single "blockchain scanner daemon", communicating with the `node-daemon` of Mintlayer, collecting information about new blocks and writing it to the master database, while having as many instances of the API web server reading from the slave databases. This ensures virtually an infinitely scalable infrastructure.
+
+### Using other database infrastructures
+
+Currently, the API server uses PostgreSQL for storage, but the design is extremely flexible and any desired database can be added if need by implementing some interface (trait) in the rust code.
+
+In addition to the PostgreSQL, an implementation of a full in-memory storage exists, which we use for testing and as a reference implementation. Hence, when adding a new database implementation, the in-memory implementation can be used as a reference implementation. Our tests ensure that both PostgreSQL and in-memory implementation, through the beautiful abstractions of rust, both arrive to the same result. Any additional implementation can be added to the same test suite.
+
```

Comment:
> the in-memory implementation can be used as a reference ~implementation~ one

To avoid tautology

> both PostgreSQL and in-memory implementation, through the beautiful abstractions of rust, ~both~ arrive to the same result

That extra `both` sounds off